Add Refactor Project Docs (#323)
FlutterFlow/flutterflow-documentation: docs/resources/projects/refactor-project.md

---
slug: refactor-project
title: Refactor Project
tags: [Refactor]
keywords: [FlutterFlow, Refactor Project, developer tools, large-scale edits, rename , key reference search, developer menu]
description: Learn how to refactor your project in FlutterFlow.
sidebar_position: 7
---

# Refactor Project

> **Refactor Project** is a developer‑focused mode that opens your FlutterFlow project as a set of YAML files so you can perform large-scale edits, such as renaming data types, pages, enums, or components, in a single, consistent operation.

Reviewer: Is is just renaming? Can we provide a list of refactoring possible through this method presented in a bullet list?

Author: Yup, added them as a "Possible Use Cases" section.

> For example, if you want to use a custom data type from a Library and update all references, you don’t have to manually edit each page or component. With Refactor Project, you can update all references in one go using a single refactor pass.

Reviewer: ```suggestion
For example, if you want to use a custom data type from a Library and update all references, you don’t have to manually edit each page or component. With this mode enabled, you can update all references at once using a single refactor pass.
```

> It makes managing large projects easier and more reliable. You can make changes across hundreds of references in just seconds, which saves time and effort compared to manually editing. It also lets you preview changes and dismiss anything you don't want to update.

Reviewer: ```suggestion
It makes managing large projects easier and more reliable. You can make changes across hundreds of references in just seconds, saving time and effort compared to manual edits. It also lets you preview changes and dismiss anything you don’t want to update.
```

:::info

> You can refactor the project only if you're on a [**paid plan**](https://www.flutterflow.io/pricing).

Reviewer: All paid plans?

:::

To refactor a project, go to **Toolbar > Developer Menu > Refactor Project**. You’ll need to commit any unsaved changes before entering the refactor view. This opens your project in a YAML-based editor, where you can search, edit, and replace values across multiple files.

> You can also use **key reference** search by toggling the **key** icon—currently supported for data types, enums, pages, and components. Changes are color-coded: added lines appear green, and removed lines appear red. As you make changes, FlutterFlow provides inline YAML validation to help you catch and fix any issues in real time.

Reviewer: ```suggestion
You can also use **key reference** search by toggling the **key** icon—currently supported for data types, enums, pages, and components. Changes are color-coded: added lines appear in green, and removed lines appear in red. As you make changes, FlutterFlow provides inline YAML validation to help you catch and fix issues in real time.
```
